src/spa/src/simple_parser/parser.h

```diff
@@ -12,17 +12,17 @@ namespace Simple {
 class Expr {
  public:
   virtual ~Expr(){};
-  virtual bool operator==(const Expr& other) const = 0;
+  virtual bool operator==(const Expr &other) const = 0;
   virtual std::string str() = 0;
 };
 
 // NumberExprAST - Expression class for numeric literals
 class NumberExpr : public Expr {
  public:
   std::string Val;
-  NumberExpr(const std::string& Val) : Val(Val){};
-  bool operator==(const Expr& other) const {
-    auto casted_other = dynamic_cast<const NumberExpr*>(&other);
+  NumberExpr(const std::string &Val) : Val(Val){};
+  bool operator==(const Expr &other) const {
```

Comment:
These shifting of pointer thingies to the start of the variable doesn't happen for me, I think the new Google style prefers it the other way around.